src/redux/modules/submissions.js

```diff
@@ -143,17 +143,21 @@ const reducer = handleActions(
                   'data',
                   data =>
                     itemId === id
-                      ? data.set('accepted', true)
-                      : data.set('accepted', false)
+                      ? data
+                          .set('accepted', true)
+                          .set('accepted-pending', false)
+                      : data
+                          .set('accepted', false)
+                          .set('accepted-pending', false)
                 )
               : item
         )
       ),
     [additionalActionTypes.UNACCEPT_PENDING]: (state, { meta: { id } }) =>
-      state.setIn(['resources', id, 'data', 'accepted'], false),
```

Comment:
Well, this optimistic update (setting accepted to false) could have been kept (as long as there is accepted-pending set). But it is not that important.